flink-table-store-core/src/main/java/org/apache/flink/table/store/table/sink/TableCommitImpl.java

```diff
@@ -0,0 +1,137 @@
+    @Nullable private Lock lock;
+
+    public TableCommitImpl(
+            FileStoreCommit commit,
+            @Nullable FileStoreExpire expire,
+            @Nullable PartitionExpire partitionExpire) {
+        this.commit = commit;
+        this.expire = expire;
+        this.partitionExpire = partitionExpire;
+    }
+
+    @Override
+    public TableCommitImpl withOverwritten(
+            @Nullable List<Map<String, String>> overwritePartitions) {
+        this.overwritePartitions = overwritePartitions;
+        return this;
+    }
+
+    @Override
+    public TableCommitImpl withLock(Lock lock) {
+        commit.withLock(lock);
+
+        if (expire != null) {
+            expire.withLock(lock);
+        }
+
+        if (partitionExpire != null) {
+            partitionExpire.withLock(lock);
+        }
+
+        this.lock = lock;
+        return this;
+    }
+
+    @Override
+    public TableCommitImpl ignoreEmptyCommit(boolean ignoreEmptyCommit) {
+        commit.ignoreEmptyCommit(ignoreEmptyCommit);
+        return this;
+    }
+
+    @Override
+    public Set<Long> filterCommitted(Set<Long> commitIdentifiers) {
+        return commit.filterCommitted(commitIdentifiers);
+    }
+
+    @Override
+    public void commit(long identifier, List<CommitMessage> commitMessages) {
+        ManifestCommittable committable = new ManifestCommittable(identifier);
+        for (CommitMessage commitMessage : commitMessages) {
+            committable.addFileCommittable(commitMessage);
+        }
+        commit(Collections.singletonList(committable));
+    }
+
+    public void commit(List<ManifestCommittable> committables) {
+        if (overwritePartitions == null) {
+            for (ManifestCommittable committable : committables) {
+                commit.commit(committable, new HashMap<>());
+            }
+        } else {
+            ManifestCommittable committable;
+            if (committables.size() > 1) {
+                throw new RuntimeException(
+                        "Multiple committables appear in overwrite mode, this may be a bug, please report it: "
+                                + committables);
+            } else if (committables.size() == 1) {
+                committable = committables.get(0);
+            } else {
+                // create an empty committable
+                // identifier is Long.MAX_VALUE, come from batch job
+                // TODO maybe it can be produced by CommitterOperator
+                committable = new ManifestCommittable(Long.MAX_VALUE);
+            }
+            commit.overwrite(overwritePartitions, committable, new HashMap<>());
+        }
+
+        if (expire != null) {
+            expire.expire();
+        }
+
+        if (partitionExpire != null) {
+            partitionExpire.expire();
+        }
+    }
+
+    @Override
+    public void close() throws Exception {
+        if (lock != null) {
+            lock.close();
```

Comment:
`lock` sets to null.

Reply:
I think let it non-null to throw exception.